src/classes/SftpSessionHandler.ts

```diff
@@ -256,6 +256,8 @@ export class SftpSessionHandler {
       temporaryFile.fd,
       data,
       0,
+      data.length,
+      offset,
```

Comment:
Before merging I want to understand what happens if offset is an incredibly high number.

We may need to add some type of validation here.

Reply:
My understanding is that the offset will result in a zero-filled file up to the offset, so if offset was 1TB that would create a 1TB file.

We have a decision to make!

Reply:
OK so we had some discussion during today's rclone checkin.

Karl raised the idea of simply not allowing non-sequential writes (throwing an error / returning a failure status if offsets are not sequential).  This would prevent corruption but it might result in unpredictable end user experiences depending on whether a given SFTP client attempted non-sequential writes.

That said, I will spend a moment looking at whether or not rclone runs in sequence at all times.

The "right" solution is to (1) fix our temporary storage problem entirely to remove the threat and (2) make it possible for SFTP service to know a user's quota and enforce that quota.  The first step is already captured in #145 I'll make an issue for the second.

Reply:
In the mean time I added in the 5 GB per-file limit, which feels at least a little bit safer.